AK Zeus/app/src/main/java/ayds/zeus/songinfo/moredetails/view/MoreDetailsView.kt

```diff
@@ -60,61 +61,98 @@ class MoreDetailsViewActivity : AppCompatActivity(), MoreDetailsView {
         moreDetailsModel = MoreDetailsModelModule.getMoreDetailsModel()
     }
 
+
     private fun initProperties() {
         cardDescriptionTextPane = findViewById(R.id.textPane2)
         viewFullArticleButton = findViewById(R.id.openUrlButton)
         sourceLogoImageView = findViewById(R.id.imageView)
         cardSourceTextPane = findViewById(R.id.sourcePane)
+        sourceSpinner = findViewById(R.id.dataSourceSpinner)
     }
 
+    private fun initSourceSpinner(){
+        val sources = ArrayList<String>()
+        if(!uiState.cardsStateList.isEmpty()){
+            loadSpinner(sources)
+        }
+        else{
+            sourceSpinner.setVisibility(View.GONE);
+        }
+    }
+
+    private fun loadSpinner(sources: MutableList<String> ){
+        for(card in uiState.cardsStateList){
```

Comment:
```suggestion
        val sources = uiState.cardsStateList.map{ it.source.toString() }
```

 fijense que estaban inicializando la variable en un metodo y seteandolo en el otro